core/vm/instructions.go

```diff
@@ -486,7 +486,13 @@ func opBlockhash(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) (
 	if num64 >= lower && num64 < upper {
 		// if Prague is active, read it from the history contract (EIP 2935).
 		if evm.chainRules.IsPrague {
-			num.SetBytes(getBlockHashFromContract(num64, evm.StateDB, evm.Accesses).Bytes())
+			blockHash, statelessGas := getBlockHashFromContract(num64, evm.StateDB, evm.Accesses)
+			if num64 != upper-1 && interpreter.evm.chainRules.IsEIP4762 {
```

Comment:
Note that here I'm separating `IsEIP4762` from an assumption that that is included in `IsPrague`. Not sure if makes sense to do that, but just mentioning.

Reply:
yes it does, because we are pushing for eip2935 to be included in Prague.